sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaClassExtensionScram.java

```diff
@@ -0,0 +1,89 @@
+package com.salesforce.kafka.test;
+
+import com.salesforce.kafka.test.junit5.SharedKafkaTestResource;
+import com.salesforce.kafka.test.junit5.SharedZookeeperTestResource;
+import com.salesforce.kafka.test.listeners.BrokerListener;
+import com.salesforce.kafka.test.listeners.SaslScramListener;
+import java.lang.reflect.Field;
+import kafka.admin.ConfigCommand;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * Creates and starts up a {@link SharedKafkaTestResource} that uses a custom {@link
+ * SharedZookeeperTestResource} with installed SCRAM-SHA-* credentials.
+ *
+ * <p>See {@link SharedKafkaTestResource} for usage.
+ */
+public class SharedKafkaClassExtensionScram extends SharedKafkaTestResource {
+  private final SharedZookeeperTestResource sharedZookeeperTestResource =
+      new SharedZookeeperTestResource();
+
+  @Override
+  protected void setKafkaCluster(KafkaCluster kafkaCluster) {
+    // we need to replace the zkTestServer in the kafka cluster in order to change the Zookeper
+    // configuration after it started and before the kafka broker starts.
+    if (kafkaCluster instanceof KafkaTestCluster) {
+      try {
+        Field f = KafkaTestCluster.class.getDeclaredField("zkTestServer");
+
+        // the field is private, so we need to make it accessible
+        f.setAccessible(true);
+
+        // use the already started (and configured) zookeeper server.
+        f.set(kafkaCluster, sharedZookeeperTestResource.getZookeeperTestServer());
+      } catch (Exception e) {
+        throw new RuntimeException("Error on replacing the zkTestServer", e);
+      }
+    }
+
+    super.setKafkaCluster(kafkaCluster);
+  }
+
+  @Override
+  public void beforeAll(ExtensionContext context) throws Exception {
+    // First start Zookeeper to be able to change the configuration
+    sharedZookeeperTestResource.beforeAll(context);
+    configureScramUser("admin", "admin-secret");
+    configureRegisteredUser();
+    super.beforeAll(context);
+  }
+
+  private void configureRegisteredUser() {
```

Comment:
I would move this method down for better ordering (next to `configureScramUser`)